streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfiguration.java

```diff
@@ -359,6 +359,10 @@ private AssignmentConfigs(final StreamsConfig configs) {
                           final Integer maxWarmupReplicas,
                           final Integer numStandbyReplicas,
                           final Long probingRebalanceIntervalMs) {
+            if (maxWarmupReplicas < 1) {
+                throw new IllegalArgumentException("must configure at least one warmup replica");
+            }
+
```

Comment:
Should we add the same for all the other configs?

Reply:
We could; to be fair, the config parser already checks the limits, so it's not necessary for production code.

It only comes up when we manually create these internal objects for tests. I added this particular bound specifically because I had previously passed in what I thought was a dummy value, which turned out to be a misconfiguration that actually affected the behavior I was testing.

Offhand, it doesn't seem like the same thing would happen with probingRebalanceIntervalMs, so it doesn't seem like the check here would have the same benefit; WDYT?

Reply:
Nevermind; I refactored the class to use the same config validation for passed-in arguments.